src/Kriti/Eval.hs

```diff
@@ -84,8 +104,8 @@ typoOfJSON J.Number {} = "Number"
 typoOfJSON J.Bool {} = "Boolean"
 typoOfJSON J.Null = "Null"
 
-eval :: ValueExt -> ExceptT EvalError (Reader Ctxt) J.Value
-eval = \case
+evalWith :: Map.HashMap T.Text (J.Value -> Either EvalError J.Value) -> ValueExt -> ExceptT EvalError (Reader Ctxt) J.Value
```

Comment:
If we do the function lookup during parsing and pass the actual haskell function in `Function` then we can treat it exactly the same as other kriti terms and not split up the evaluator like this.

Reply:
Adding to note and marking as resolved for now.